openpiv/pyprocess.py

```diff
@@ -340,14 +344,18 @@ def find_all_second_peaks(corr, width = 2):
     ifin[ifin > corr.shape[1]] = corr.shape[1]
     jini[jini < 0] = 0
     jfin[jfin > corr.shape[2]] = corr.shape[2]
-    # create a masked view of the corr
-    tmp = corr.view(np.ma.MaskedArray)
+    # create a masked view of the corr, is too slow?
+    #tmp = corr.view(np.ma.MaskedArray)
+    tmp = corr.copy()
     for i in ind:
-        tmp[i, iini[i]:ifin[i], jini[i]:jfin[i]] = np.ma.masked
+        tmp[i, iini[i]:ifin[i], jini[i]:jfin[i]] = 0 #ma.masked 
```

Comment:
i'm not sure but I think 0 might cause issues with non-clipped correlation maps. we had in the past some negative correlations and i think it's a reason to use np.ma.masked

Reply:
I'll check other solutions, but  not using numpy masked arrays were up to 3x faster for large sliding window arrays. I do see a severe issue with negative correlation values, so I'll go ahead to do more testing for a better solution before we go back to masked arrays.

Reply:
maybe we use np.ma.masked incorrectly. I think we shall try to make `tmp = np.ma.array()` first and then just add mask whenever it's needed. 
how do you test for memory use? I'll try this solution

Reply:
For memory usage on numpy arrays, I use np.nbytes() and divide by 1e6 to get MB.